app/models/source.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class Source
+  USER = 'user'
+  SCHOLARSPHERE = 'scholarsphere'
+  OPEN_ACCESS_BUTTON = 'open_access_button'
+  UNPAYWALL = 'unpaywall'
+  DICKINSON_IDEAS = 'dickinson_ideas'
+  PSU_LAW_ELIBRARY = 'psu_law_elibrary'
+
+  def initialize(source)
+    @source = source
+  end
+
+  def ==(other)
+    to_s == other.to_s
+  end
+  alias :eql? :==
+
+  def display
+    I18n.t(
+      @source,
```

Comment:
Would this be nil if the attribute wasn't present in OpenAccessLocation?

Reply:
It would be `""` but I could easily make it nil.

Given that these are used in enums, which do not allow it to be nil, this is super unlikely to happen in practice, but yes you could force it to happen